### PR TITLE
Create a version of cherrypicker supporting the gardener release note format

### PR DESCRIPTION
**Pull request by oliver-goetz**
/kind enhancement

**What this PR does / why we need it**:
Gardener is annotation release notes with `(breaking|feature|bugfix|doc|other) (user|operator|developer|dependency)` which is different than Kubernetes which is using `release-note`.
Because of this cherrypicker is not able to copy release notes into cherrypick PRs.

The regexp identifying the release note is not configurable yet. We forked the [cherrypicker external prow plugin](https://github.com/kubernetes/test-infra/tree/master/prow/external-plugins/cherrypicker) to change it. Eventually, it might become configurable and we could delete this fork again.

I would like to keep the effort low. Thus, the only things which are changes are the mentioned regexp, corresponding test cases and a minor fix to satisfy golangci-lint.

**Special notes for your reviewer**:
When this PR is merged and the image is build I will create a second PR to switch to the resulting image in [cherrypicker deployment](https://github.com/gardener/ci-infra/blob/5768d2ac0dd279d18bde17baf8ba89624a587fd2/config/prow/cluster/cherrypicker_deployment.yaml#L28)

### CODE REVIEW

**Comment by timebertt** [comment]
> The failing `pull-ci-infra-prow-go-tests` is a chicken or the egg dilemma I updated all dependencies to the latest version which requires go 1.18. I update the test job in this PR too, but in the config used by prow it is still running on the go 1.17 and is failing because of that.
> 
> To solve this dilemma start `pull-ci-infra-prow-go-tests` with the new go 1.18 image manually.

You could also bump the job images in a separate PR. Building 1.17 code should generally work with 1.18 images :)

File: prow/external-plugins/cherrypicker/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Cherrypicker
+
+**Note: This is a fork of [cherrypicker from kubernetes/test-infra](https://github.com/kubernetes/test-infra/tree/master/prow/external-plugins/cherrypicker) repository.
```
Comment (timebertt) [inline prow/external-plugins/cherrypicker/README.md:3]: Please add a note on which which commit this fork is based. Makes things easier in the long-run :)

File: prow/external-plugins/cherrypicker/server.go [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "cherrypick"
+const defaultLabelPrefix = "cherrypick/"
+
+var cherryPickRe = regexp.MustCompile(`(?m)^(?:/cherrypick|/cherry-pick)\s+(.+)$`)
+var releaseNoteRe = regexp.MustCompile("(?s)(```(breaking|feature|bugfix|doc|other) (user|operator|developer|dependency)\\s(.+?)```)")
```
Comment (timebertt) [inline prow/external-plugins/cherrypicker/server.go:46]: Can you extract your patches into a dedicated commit, so that one can see (and re-apply if necessary) what you have changed?

**Review by timebertt — COMMENTED** [review]
Very nice!
Some minor suggestions inline :)

**Comment by oliver-goetz** [comment]
> > The failing `pull-ci-infra-prow-go-tests` is a chicken or the egg dilemma I updated all dependencies to the latest version which requires go 1.18. I update the test job in this PR too, but in the config used by prow it is still running on the go 1.17 and is failing because of that.
> > To solve this dilemma start `pull-ci-infra-prow-go-tests` with the new go 1.18 image manually.
> 
> You could also bump the job images in a separate PR. Building 1.17 code should generally work with 1.18 images :)

Good idea, I created https://github.com/gardener/ci-infra/pull/289 for that.

File: prow/external-plugins/cherrypicker/server.go (hunk 2 above)
Comment (oliver-goetz) [reply prow/external-plugins/cherrypicker/server.go:46]: done

**Comment by oliver-goetz** [comment]
@timebertt I applied the changes, in order to make `pull-ci-infra-prow-go-tests` work we have to merge https://github.com/gardener/ci-infra/pull/289 first

**Review by timebertt — APPROVED** [review]
Thank you!
/lgtm
/test pull-ci-infra-prow-go-tests

**Comment by oliver-goetz** [comment]
@timebertt had to rebase the PR. Not the test succeeds
